### PR TITLE
CcdbApi - Add handling multiple addresses

### PR DESCRIPTION
**Pull request by mucharafal**
Add hosts pool.

To use hosts pool, string parameter with address should has hosts addresses separated with commas or semicolons.

It is initial implementation, it must be discussed and improved (but preferably in next PRs).

Methods, which behaviour should be discussed:
- delete - for now it iterates over all addresses and delete object everywhere
- truncate - like above
- list - take first response. As I discussed with Costin, it should rather ask all hosts and merge response. Is it correct, or current semantics is fine?

For others methods looping over addresses in case error (currently `404`, change to above `400`?) seems reasonable for me. But please, review it deeply, I am not C++ expert ;).

If you prefer, I can split changes into smaller PRs - just let me know.

### CODE REVIEW

**Comment by costing** [comment]
Hi @mucharafal ,

Thank you for the PR. I think this addresses the point of falling back between endpoints/specific storages.

Yes, everything above `400` should be considered an error and the fallback should apply.

As for the listing (`/browse/` or `/latest/`), this is a bit trickier since ideally it would reflect what the fallback would do. Maybe it's as simple as iterating the endpoints in reverse order? And for `/latest/` overriding the existing objects in the same path, while for `/browse/` ... append to the list? This can be left for a future PR though.

Cheers,

.costin

**Review by costing — DISMISSED** [review]

**Review by ktf — CHANGES_REQUESTED** [review]
It looks like the code checker is not happy?

```
/sw/SOURCES/O2/7680-slc8_x86-64/0/CCDB/src/CcdbApi.cxx:41:10: error: inclusion of deprecated C++ header 'stdio.h'; consider using 'cstdio' instead [modernize-deprecated-headers]
/sw/SOURCES/O2/7680-slc8_x86-64/0/CCDB/src/CcdbApi.cxx:486:42: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/7680-slc8_x86-64/0/CCDB/src/CcdbApi.cxx:488:15: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/7680-slc8_x86-64/0/CCDB/src/CcdbApi.cxx:1297:19: error: use nullptr [modernize-use-nullptr]
/sw/SOURCES/O2/7680-slc8_x86-64/0/CCDB/src/CcdbApi.cxx:1299:20: error: use nullptr [modernize-use-nullptr]
/sw/SOURCES/O2/7680-slc8_x86-64/0//CCDB/test/testCcdbApiMultipleUrls.cxx:1:1: error: missing or malformed copyright notice
```

could you please fix them?

File: CCDB/test/testCcdbApiMultipleUrls.cxx
Comment (Barthelemy) [inline CCDB/test/testCcdbApiMultipleUrls.cxx:16]: That's a lot of includes. Are they all necessary ?

**Review by Barthelemy — DISMISSED** [review]

File: CCDB/test/testCcdbApiMultipleUrls.cxx
Comment (mucharafal) [reply CCDB/test/testCcdbApiMultipleUrls.cxx:16]: No, you are right. I have left necessary

**Comment by costing** [comment]
@ktf could this be merged?

**Review by ktf — DISMISSED** [review]

**Comment by ktf** [comment]
I think it required an  approval before the tests could start, because @mucharafal is a first time contributor. Let's wait for the tests and then we can merge.

**Comment by mucharafal** [comment]
> I think it required an approval before the tests could start, because @mucharafal is a first time contributor. Let's wait for the tests and then we can merge.

@ktf When the test will be started? What is the algorithm?

**Comment by ktf** [comment]
Some of them already started. It should update things shortly.

**Review by sawenzel — APPROVED** [review]
triggering build

**Comment by sawenzel** [comment]
Merging. I would still be good to put (doxygen) documentation on the new functions introduced as well as to enhance the main markdown README with the features.

**Pull request merged**
